server/resources/GmailResources.py

```diff
@@ -0,0 +1,159 @@
+from apiclient.discovery import build
+from oauth2client.client import FlowExchangeError
+from oauth2client.client import OAuth2Credentials
+from oauth2client.client import flow_from_clientsecrets
+from flask_restful import Resource, reqparse
+from flask_jwt_extended import jwt_required, get_jwt_identity
+import logging
+import httplib2
+import pickle
+
+from models.GmailCredentials import GmailCredentials
+from models.UserModel import UserModel
+
+authorize_parser = reqparse.RequestParser()
+authorize_parser.add_argument('code', required=True)
+
+CLIENTSECRETS_LOCATION = 'instance/client_secrets.json'
+REDIRECT_URI = 'http://localhost:5000/'
```

Comment:
Can we try redirect this to 3000, and try doing this on the react front-end?

Reply:
We would just need a react route (maybe /gmail/authorize) and then to do a javascript API call to the back-end with the authorization code (adding the jwt toekn from localstorage)